eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -1,25 +1,81 @@
+[CmdletBinding(SupportsShouldProcess)]
 param(
   # The repo owner: e.g. Azure
   $RepoOwner,
   # The repo name. E.g. azure-sdk-for-java
   $RepoName,
   # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
   $RepoId="$RepoOwner/$RepoName",
-  [Parameter(Mandatory = $true)]
-  $BranchPrefix,  
+  # CentralRepoId the original PR to generate sync PR. E.g Azure/azure-sdk-tools for eng/common
+  $CentralRepoId,
+  # We start from the sync PRs, use the branch name to get the PR number of central repo. E.g. sync-eng/common-(<branchName>)-(<PrNumber>). Have group name on PR number.
+  $CentralPRRegex,
   # Date format: e.g. Tuesday, April 12, 2022 1:36:02 PM. Allow to use other date format.
   [AllowNull()]
   [DateTime]$LastCommitOlderThan,
+  [switch]$skipCentralPR,
   [Parameter(Mandatory = $true)]
   $AuthToken
 )
 
 . (Join-Path $PSScriptRoot common.ps1)
 
+function RetrievePRAndClose {
+  [CmdletBinding(SupportsShouldProcess)]
+  param (
+    $pullRequestNumber
+  )
+
+  if (!$pullRequestNumber) {
+    return $false
+  }
+  try {
+    Write-Host "The repo [ $CentralRepoId ] pull request number is [ $pullRequestNumber ] "
+    $centralPR = Get-GitHubPullRequest -RepoId $CentralRepoId -PullRequestNumber $pullRequestNumber -AuthToken $AuthToken
+    if ($centralPR.state -ne "closed") {
+      return $true
+    }
+  }
+  catch 
+  {
+    LogError "Get-GitHubPullRequests failed with exception:`n$_"
+    return $false
+  }
+  
+  try {
+    $head = "${RepoId}:${branchName}"
+    LogDebug "Operating on branch [ $branchName ]"
+    $pullRequests = Get-GitHubPullRequests -RepoId $RepoId -State "all" -Head $head -AuthToken $AuthToken
+  }
+  catch
+  {
+    LogError "Get-GitHubPullRequests failed with exception:`n$_"
+    exit 1
+  }
+  $openPullRequests = $pullRequests | ? { $_.State -eq "open" }
+
+  foreach ($openPullRequest in $openPullRequests) {
+    try 
+    {
+      if (!$hasCentralPROpened -and $PSCmdlet.ShouldProcess("[ $($openPullRequest.url)] with branch [ $branchName ] in [ $RepoId ]", "Closing the pull request")) {
```

Comment:
This is never getting set until this function returns. We should completely skip this entire loop if the central PR is open.